components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -169,45 +202,37 @@ def run_clp(
     compression_successful = False
     proc = subprocess.Popen(compression_cmd, stdout=subprocess.PIPE, stderr=stderr_log_file)
 
-    sql_adapter = SQL_Adapter(Database.parse_obj(clp_metadata_db_connection_config))
-    with closing(sql_adapter.create_connection(True)) as db_conn, closing(
-        db_conn.cursor(dictionary=True)
-    ) as db_cursor:
-        # Compute the total amount of data compressed
-        last_archive_stats = None
-        total_uncompressed_size = 0
-        total_compressed_size = 0
-        while True:
-            line = proc.stdout.readline()
-            if not line:
-                break
-            stats = json.loads(line.decode("ascii"))
-            if last_archive_stats is not None and stats["id"] != last_archive_stats["id"]:
-                # We've started a new archive so add the previous archive's last
-                # reported size to the total
-                total_uncompressed_size += last_archive_stats["uncompressed_size"]
-                total_compressed_size += last_archive_stats["size"]
-                if tag_ids is not None:
-                    update_tags(
-                        db_conn,
-                        db_cursor,
-                        clp_metadata_db_connection_config["table_prefix"],
-                        last_archive_stats["id"],
-                        tag_ids,
-                    )
-            last_archive_stats = stats
-        if last_archive_stats is not None:
-            # Add the last archive's last reported size
-            total_uncompressed_size += last_archive_stats["uncompressed_size"]
-            total_compressed_size += last_archive_stats["size"]
+    # Compute the total amount of data compressed
+    last_archive_id = ""
+    total_uncompressed_size = 0
+    total_compressed_size = 0
+    while True:
+        line = proc.stdout.readline()
+        if not line:
+            break
+        stats = json.loads(line.decode("ascii"))
+        if stats["id"] != last_archive_id:
```

Comment:
Hm, since an archive may print its stats multiple times (after every segment is created in `clp`'s case), then we would still need the previous logic that keeps `last_archive_stats`, right?

Reply:
The previous logic is to check last_archive_stats['id'], if it's different from current id, we update everything. That's the same as current logic?

Reply:
Not exactly. In the previous logic if `last_archive_stats['id']` was different from the current ID, we would add `uncompressed_size` and `size` from `last_archive_stats`, but in the current code, we're adding the values from `stats`.

To see why this is a problem, imagine `clp` creates two archives with two segments each, meaning it will print the archive stats 4 times, something like this:

1. archive-1-seg-1: `uncompressed_size = 10, size = 1`
2. archive-1-seg-2: `uncompressed_size = 20, size = 2`
3. archive-2-seg-1: `uncompressed_size = 5, size = 1`
4. archive-2-seg-2: `uncompressed_size = 10, size 2`

In the current code, when we see the printout of (1), we will do `total_uncompressed_size += 10, size += 1`. When we see the printout of (3), we will do `total_uncompressed_size += 5, size += 1`. This will give us `total_uncompressed_size = 15, size = 2`. But it should be `total_uncompressed_size = 30, size = 4`.

Reply:
Oh I see your point. `uncompressed_size` is cumulative. I thought we can accept the first one, and abandon the rest with the same id.